commit 9da66f8898a69b77121a4f2ae785e9cd29e94e6c
Author: Yang Bai <hamo.by@gmail.com>
Date:   Tue Oct 18 11:53:11 2011 -0400

    mount.nfs: Mount should really return from errno test
    
    We should only try next address family if we meet ECONNREFUSED or
    EHOSTUNREACH for v4 or ECONNREFUSED or EOPNOTSUPP or EHOSTUNREACH for v3v2.
    Before, only a break in swich can not make the program out of for loop.
    
    Signed-off-by: Yang Bai <hamo.by@gmail.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/utils/mount/stropts.c b/utils/mount/stropts.c
index 314a806..4032bf3 100644
--- a/utils/mount/stropts.c
+++ b/utils/mount/stropts.c
@@ -665,9 +665,10 @@ static int nfs_try_mount_v3v2(struct nfsmount_info *mi)
 		case EHOSTUNREACH:
 			continue;
 		default:
-			break;
+			goto out;
 		}
 	}
+out:
 	return ret;
 }
 
@@ -751,9 +752,10 @@ static int nfs_try_mount_v4(struct nfsmount_info *mi)
 		case EHOSTUNREACH:
 			continue;
 		default:
-			break;
+			goto out;
 		}
 	}
+out:
 	return ret;
 }
 
